src/app/locator/qgsinbuiltlocatorfilters.cpp

```diff
@@ -298,6 +302,108 @@ void QgsActiveLayerFeaturesLocatorFilter::triggerResult( const QgsLocatorResult
   QgisApp::instance()->mapCanvas()->zoomToFeatureIds( layer, QgsFeatureIds() << id );
 }
 
+//
+// QgsAllLayersFeaturesLocatorFilter
+//
+
+QgsAllLayersFeaturesLocatorFilter::QgsAllLayersFeaturesLocatorFilter( QObject *parent )
+  : QgsLocatorFilter( parent )
+{
+  setUseWithoutPrefix( false );
+}
+
+QgsAllLayersFeaturesLocatorFilter *QgsAllLayersFeaturesLocatorFilter::clone() const
+{
+  return new QgsAllLayersFeaturesLocatorFilter();
+}
+
+void QgsAllLayersFeaturesLocatorFilter::prepare( const QString &string, const QgsLocatorContext & )
+{
+  if ( string.length() < 3 )
+    return;
+
+  const QMap<QString, QgsMapLayer *> layers = QgsProject::instance()->mapLayers();
+  for ( auto it = layers.constBegin(); it != layers.constEnd(); ++it )
+  {
+    QgsVectorLayer *layer = qobject_cast< QgsVectorLayer *>( it.value() );
+    if ( !layer || !layer->searchable() )
```

Comment:
I'm not sure if it's too late for this - but I don't think searchable() should be in QgsVectorLayer API. I think this should instead be a customProperty on the layer to avoid complicating the api too much.

Reply:
Already discussed this with Matthias about the geometry stuff and I am not sure the cluttering of the API is more important that the readability of code and the ability of easily finding the methods.
Anyway, regarding this point, I have open a question here
https://github.com/qgis/QGIS-Enhancement-Proposals/issues/125#issuecomment-418755712
question F, the last paragraph of my comment:

> In project properties, we can define for each layer:
> * identifiable (saved in project as non-identifiable list)
> * read-only (saved in layer)
> * searchable (saved in layer)
> * non-removable (saved in project)
> Being saved in the project means it is not written in the style definition (XML).
While it makes sense for non-removable tag, I am not really sure about the 3 others.
We should have coherent way of saving these 3 values.
Shall they be seen as layer properties (as style) or as project properties?

What do you think?
If we move to project, that closes this case.

Reply:
I think they still belong in layer properties, since they apply to layers and not a project. What do you think about making identifiable and searchable as flags? That would seem nice and clean to me

Reply:
implemented in https://github.com/qgis/QGIS/pull/7815